### DISCUSSION TITLE
Is wsl2 fast enough with external hdd?

### DISCUSSION THREAD

**Discussion by weeebdev** [microsoft/WSL#6109] [Q&A] [answered]
I am planning to move all of my projects to external hdd, since I don't have memory on my laptop. Should I buy external hdd for it and how it will affect the speed? I know that speed performance on windows filesystem is very slow on WSL2. Thank you.

**Comment by craigloewen-msft** [comment, marked as answer]
You can solve this by putting your WSL distro on that external HDD! Here are the sample steps: 

* Get a distro .tar (I'd just use `wsl --export <distroName>` on a new instance of a distro)
* Run `wsl --import <distroName> <installLocation> <FileName>` and set your install location to your external HDD
* Set your user in the distro using [this value in /etc/wsl.conf](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#user)
* Run your distro!

**Reply by weeebdev** [reply]
I think, it won't be so fast on the ntfs+hdd connected via usb, I guess?

**Reply by craigloewen-msft** [reply]
Ah, yes that will be much slower since the actual read and write calls for that set up will be slower.

**Reply by LuchoRod** [reply]
when i try to import the distro using `wsl --import` i got this message and wsl does nothing.
![Screenshot 2021-09-13 142141](https://user-images.githubusercontent.com/2924830/133163946-71acc3fc-4c44-4ea3-b219-effeb2ab98b8.png)

what do i miss?

thanks in advance!

**Comment by Biswa96** [comment]
By "external HDD" if you mean to connect with a USB cable then I wouldn't do that. I would buy a SATA cable and connect HDD/SSD with motherboard.

**Reply by weeebdev** [reply]
Unfortunately, I am not able to do it on my ultra 😢

**Comment by weeebdev** [comment]
@Biswa96 @craigloewen-msft   what if I keep wsl on my ssd along with windows, but write/save projects in external hdd? Will it be okay? or write projects in wsl filesystem and then move it to external hdd?

**Reply by craigloewen-msft** [reply]
If you're using WSL 2 I'd recommend putting the projects that you want Linux binaries to run on in your Linux file system.
